### PR TITLE
Cleanup phpstan

### PR DESCRIPTION
**Pull request by dannyvw**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | #... 
| License       | MIT
| Doc PR        | api-platform/docs#...

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -20,14 +20,10 @@ parameters:
 		- tests/Fixtures/app/var/
 		- tests/Fixtures/Symfony/Maker
 		# BC layer
-		- src/deprecated_interfaces.php
 		- tests/Fixtures/TestBundle/BrowserKit/Client.php
```
Comment (dannyvw) [inline phpstan.neon.dist:23]: This line needs to be checked

File: phpstan.neon.dist [hunk 2]
```diff
@@ -20,14 +20,10 @@ parameters:
...
 		# The Symfony Configuration API isn't good enough to be analysed
 		- src/Symfony/Bundle/DependencyInjection/Configuration.php
-		# Phpstan runs on phpunit > 9, a signature changed in this file
-		- src/Symfony/Bundle/Test/Constraint/ArraySubsetLegacy.php
 		# Imported code (temporary)
-		- src/Symfony/Bundle/Test/BrowserKitAssertionsTrait.php
 		- tests/Symfony/Bundle/Test/WebTestCaseTest.php
```
Comment (dannyvw) [inline phpstan.neon.dist:27]: This line needs to be checked

File: phpstan.neon.dist [hunk 3]
```diff
@@ -20,14 +20,10 @@ parameters:
...
 		- tests/Fixtures/TestBundle/Security/AbstractSecurityUser.php
```
Comment (dannyvw) [inline phpstan.neon.dist:28]: This line needs to be checked

File: phpstan.neon.dist (hunk 2 above)
Comment (alanpoulain) [reply phpstan.neon.dist:25]: This PR is a WIP then?

Comment (dannyvw) [reply phpstan.neon.dist:25]: We can do it in this PR or in a new one. I think this class can be removed but I'm not sure.

File: phpstan.neon.dist (hunk 3 above)
Comment (dannyvw) [reply phpstan.neon.dist:26]: Same for this. Can the BC layer be removed?

File: phpstan.neon.dist (hunk 2 above)
Comment (alanpoulain) [reply phpstan.neon.dist:25]: I will do it in another PR :slightly_smiling_face:

Comment (alanpoulain) [reply phpstan.neon.dist:25]: Done here: https://github.com/api-platform/core/pull/4866

File: phpstan.neon.dist (hunk 3 above)
Comment (alanpoulain) [reply phpstan.neon.dist:26]: Yes, can you do it in another PR?

Comment (dannyvw) [reply phpstan.neon.dist:26]: https://github.com/api-platform/core/pull/4867

**Pull request merged**